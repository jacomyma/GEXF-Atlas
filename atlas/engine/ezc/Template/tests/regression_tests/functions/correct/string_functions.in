1. {str_replace( "Say yes!", 4, 3, "no")}
2. {str_remove( "Say yes!", 4, 3 )}
3. {str_chop( "Say yes!", 5 )}
4. {str_chop_front( "Say yes!", 4 )}
5. {str_append( "abc", "def" )}
6. {str_prepend( "abc", "def" )}
7. {str_compare( "abcd", "abcd" ) == 0}{str_compare( "abc2", "abc12" ) > 0}
8. {str_nat_compare( "abc2", "abc12" ) < 0}
9. {str_contains("je moeder", "moe")}
10. {str_len("abcdefg")}

11. {str_left("abcdefg", 3)}
12. {str_starts_with("abcdefg", "abc")}
13. {str_right("abcdefg", 3)}
14. {str_ends_with("abcdefg", "efg")}
15. {str_mid( "abcdefg", 3, 2)}
16. {str_at( "abcdefg", 1)}
17. {str_fill( "ab", 3)}
18. {str_index_of( "abcdefgcdab", "def")} {str_index_of( "abcdefgcdab", "cd", 5 ) }
18.2. {str_last_index( "abcdefgcdab", "ab")} {str_last_index( "abcdefgcdab", "ab", -5 ) }
19. {str_is_empty( "" )} {str_is_empty("") !== false}
20. {str_pad_left( "abc", 5, ".." )}

21. {str_pad_right( "abc", 5, ".." )}
22. {str_number( 3.14159, 2, ",", "")}
23. #{str_trim( "  blabla    ")}# #{str_trim( ". bla", ".")}#
24. #{str_trim_left( "  blabla  ")}#
25. #{str_trim_right( "  blabla  ")}#
26. #{str_simplify( "    hello    world

bla     " )}#
27. {var $a = str_split("hello/world", "/")}{$a[1]} {$a[0]}
28. {str_join( array("hello", "world"), " ")}
29. {str_upper( "heLLo World" )}
30. {str_lower( "HelLO WoRLd" )}

31. {str_capitalize( "hello world" )}
32. {str_find_replace( "Hello earth!", "earth", "world" )}
33. {str_reverse( "Hello world" )}
34. {str_word_count( "Hello world" )}
35. {str_paragraph_count("This text

contains 

two paragraphs")}
36. {str_wrap("abcdefghijklmnop", 4, "\n", true)}{str_wrap("qrst asdf asdf", 5, "\n")} 
37. {str_base64_encode( "I don't want to be encoded!" )}
38. {str_base64_decode( "SSBkb24ndCB3YW50IHRvIGJlIGVuY29kZWQh" )}
39. {str_ord( "A" )}
40. {str_char_count( "hello" )}
41. {str_index_of( "Hello", "l")}
42. {str_chr( 65 )}
