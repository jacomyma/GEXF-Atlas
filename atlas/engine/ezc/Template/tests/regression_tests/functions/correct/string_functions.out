1. Say no!
2. Say !
3. Say
4. yes!
5. abcdef
6. defabc
7. 11
8. 1
9. 1
10. 7

11. abc
12. 1
13. efg
14. 1
15. de
16. b
17. ababab
18. 3 7
18.2. 9 0
19. 1 1
20. ..abc

21. abc..
22. 3,14
23. #blabla# # bla#
24. #blabla  #
25. #  blabla#
26. #hello world bla#
27. world hello
28. hello world
29. HELLO WORLD
30. hello world

31. Hello world
32. Hello world!
33. dlrow olleH
34. 2
35. 2
36. abcd
efgh
ijkl
mnopqrst
asdf
asdf 
37. SSBkb24ndCB3YW50IHRvIGJlIGVuY29kZWQh
38. I don't want to be encoded!
39. 65
40. 5
41. 2
42. A
